pkg/controller/util/propagatedversion.go

```diff
@@ -17,64 +17,45 @@ limitations under the License.
 package util
 
 import (
+	"fmt"
 	"reflect"
 	"sort"
-	"strconv"
+	"strings"
 
-	"github.com/pkg/errors"
+	"k8s.io/apimachinery/pkg/apis/meta/v1/unstructured"
 
-	"github.com/kubernetes-sigs/federation-v2/pkg/apis/core/common"
 	fedv1a1 "github.com/kubernetes-sigs/federation-v2/pkg/apis/core/v1alpha1"
-	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
 )
 
-type ComparisonHelper interface {
-	GetVersion(objectMeta metav1.Object) string
-	Equivalent(objMeta1, objectMeta2 metav1.Object) bool
-}
+const (
+	generationPrefix      = "gen:"
+	resourceVersionPrefix = "rv:"
+)
 
-// NewComparisonHelper instantiates and returns a Resource or Generation Helper
-// struct that implements the ComparisonHelper interface based on the version
-// comparison type passed in.
-func NewComparisonHelper(comparisonField common.VersionComparisonField) (ComparisonHelper, error) {
-	switch comparisonField {
-	case common.ResourceVersionField:
-		return &ResourceHelper{}, nil
-	case common.GenerationField:
-		return &GenerationHelper{}, nil
-	default:
-		return nil, errors.Errorf("Unrecognized version comparison field %v", comparisonField)
+// ObjectVersion retrieves the field type-prefixed value used for
+// determining currency of the given cluster object.
```

Comment:
`determining currency`?, is this some term. or is it currency == version?

Reply:
'determining whether the object is current, up-to-date'